dpnp/backend/custom_kernels_sorting.cpp

```diff
@@ -0,0 +1,115 @@
+
+#include <iostream>
+
+#include <backend/backend_iface.hpp>
+#include "backend_pstl.hpp"
+#include "queue_sycl.hpp"
+
+template <typename _DataType, typename _idx_DataType>
+struct _argsort_less
+{
+    _argsort_less(_DataType* data_ptr)
+    {
+        _data_ptr = data_ptr;
+    }
+
+    inline bool operator()(const _idx_DataType& idx1, const _idx_DataType& idx2)
+    {
+        return (_data_ptr[idx1] < _data_ptr[idx2]);
+    }
+
+private:
+    _DataType* _data_ptr = nullptr;
+};
+
+template <typename _DataType, typename _idx_DataType>
+class custom_argsort_c_kernel;
+
+template <typename _DataType, typename _idx_DataType>
+void custom_argsort_c(void* array1_in, void* result1, size_t size)
+{
+    _DataType* array_1 = reinterpret_cast<_DataType*>(array1_in);
+    _idx_DataType* result = reinterpret_cast<_idx_DataType*>(result1);
+
+    for (size_t i = 0; i < size; ++i)
+    {
+        result[i] = i;
+    }
+
+    auto policy =
+        oneapi::dpl::execution::make_device_policy<class custom_argsort_c_kernel<_DataType, _idx_DataType>>(DPNP_QUEUE);
+
+    std::sort(policy, result, result + size, _argsort_less<_DataType, _idx_DataType>(array_1));
+
+    policy.queue().wait();
+}
+
+template void custom_argsort_c<double, long>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<float, long>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<long, long>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<int, long>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<double, int>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<float, int>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<long, int>(void* array1_in, void* result1, size_t size);
+template void custom_argsort_c<int, int>(void* array1_in, void* result1, size_t size);
+
+template <typename _DataType>
+struct _sort_less
+{
+    inline bool operator()(const _DataType& val1, const _DataType& val2)
+    {
+        return (val1 < val2);
+    }
+};
+
+template <typename _DataType>
+class custom_sort_c_kernel;
+
+template <typename _DataType>
+void custom_sort_c(void* array1_in, void* result1, size_t size)
+{
+    _DataType* array_1 = reinterpret_cast<_DataType*>(array1_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    for (size_t i = 0; i < size; ++i)
+    {
+        result[i] = array_1[i];
+    }
+
+    auto policy = oneapi::dpl::execution::make_device_policy<class custom_sort_c_kernel<_DataType>>(DPNP_QUEUE);
+
+    // fails without explicitly specifying of comparator or with std::less during kernels compilation
+    // affects other kernels
+    std::sort(policy, result, result + size, _sort_less<_DataType>());
+
+    policy.queue().wait();
+}
+
+template void custom_sort_c<double>(void* array1_in, void* result1, size_t size);
```

Comment:
Not sure in it but it would be greate to add these kernels into FPTR https://github.com/IntelPython/dpnp/blob/9b63bae0b6db57aac85fa0b093ff8b3843b2fa5d/dpnp/backend/backend_iface_fptr.cpp#L46

Reply:
done